### PR TITLE
[ec2_snapshot] Some refactoring to follow conventions

### PR DESCRIPTION
**Pull request by prasadkatti**
##### SUMMARY

- move delete snapshot code into its own function
- use `required_if`, `required_together` and `mutually_exclusive` wherever possible
- minor changes to examples
- renamed snapshots_tags to tags. snaphot_tags is now an alias for tags.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`ec2_snapshot`

##### ANSIBLE VERSION

```
ansible 2.5.0
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_snapshot.py [hunk 1]
```diff
@@ -245,28 +225,43 @@ def create_snapshot(module, ec2, state=None, description=None, wait=None,
...
+
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
-            volume_id = dict(),
-            description = dict(),
-            instance_id = dict(),
-            snapshot_id = dict(),
-            device_name = dict(),
+            volume_id = dict(type='str'),
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/ec2_snapshot.py:246]: You don't need to document type='str' since it's the default.

**Review by s-hertel — APPROVED** [review]
This looks good to me. I wonder if the module should just be ported to boto3?

**Comment by prasadkatti** [comment]
@s-hertel - Yes, I am planning to work on porting this to boto3 :) . But I thought I would start with smaller easy-to-verify changes first.

**Comment by willthames** [comment]
To be honest, porting it to boto3 isn't the problem (#22394).

Having a working test suite is.

**Pull request closed without merging**